From 1e70899d6a58651b786b6fe7dca816d693db6f05 Mon Sep 17 00:00:00 2001
From: Stefan Eichenberger <stefan.eichenberger@toradex.com>
Date: Wed, 18 Mar 2020 19:00:10 +0100
Subject: [PATCH 3/3] arm64: dts: fsl-imx8qxp-colibri: add configuration for
 torizon

For torizon we disable all interfaces which are not part of the modules.
Here we also deactivate e.g. the dsihdmi to adapter.

Signed-off-by: Stefan Eichenberger <stefan.eichenberger@toradex.com>
---
 .../freescale/fsl-imx8qxp-colibri-eval-v3.dts |  1 +
 .../fsl-imx8qxp-colibri-torizon.dtsi          | 20 +++++++++++++++++++
 2 files changed, 21 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-torizon.dtsi

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-eval-v3.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-eval-v3.dts
index bb565ab6ea761..def3525c84dc7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-eval-v3.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-eval-v3.dts
@@ -7,6 +7,7 @@
 
 #include "fsl-imx8qxp-colibri.dtsi"
 #include "fsl-imx8qxp-colibri-eval-v3.dtsi"
+#include "fsl-imx8qxp-colibri-torizon.dtsi"
 
 / {
 	model = "Toradex Colibri iMX8QXP/DX on Colibri Evaluation Board V3";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-torizon.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-torizon.dtsi
new file mode 100644
index 0000000000000..52a9b940ef8ff
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-colibri-torizon.dtsi
@@ -0,0 +1,20 @@
+/delete-node/ &atmel_mxt_ts;
+/delete-node/ &pinctrl_mxt_ts;
+
+&iomuxc {
+	pinctrl-0 = <&pinctrl_hog0>, <&pinctrl_hog1>, <&pinctrl_hog2>,
+			<&pinctrl_ext_io0>;
+};
+
+&adma_lcdif {
+    status = "disabled";
+};
+
+&i2c0 {
+    /delete-node/ ad7879@2c;
+};
+/delete-node/ &pinctrl_ad7879_int;
+
+&ldb1 {
+    status = "disabled";
+};
-- 
2.17.1

